Converting to a BentoCTL Operator

dtemir opened the pull request:
Converting GCP Cloud Run Tool to work with `bentoctl`.

* Put everything under a module
* Added a simple `OPERATOR_SCHEMA`
* Fixed a couple things with `bentoctl` to work with the operator

```bash
$ bentoctl deploy
Interactive Deployment Spec Builder

Welcome! You are now in interactive mode.

This mode will help you setup the deployment_spec.yaml file required for
deployment. Fill out the appropriate values for the fields.

(deployment spec will be saved to: ./deployment_spec.yaml)

api_version: v1
metadata: 
    name: test
    operator: gcp-cloud-run
spec: 
    bento: .
    project_id: bentoml-43434
    region: us-central1
    port: 5000
    min_instances: 0
    max_instances: 1
    memory: 512Mi
    cpu: 1
    allow_unauthenticated: True
    platform: managed
deployment spec file exists! Should I override? [Y/n]: Y
deployment spec generated to: deployment_spec.yaml
[2021-11-21 12:20:27,634] WARNING - Saved BentoService Python version mismatch: loading BentoService bundle created with Python version 3.8.10, but current environment version is 3.7.11.
Successful deployment!```
```

to-do:
* Add more description to the Cerberus validation schema 

done:
* Need to update `describe.py` and all other scripts that use `get_configuration_value` to read from a json because `bentoctl` already passes a proper configuration dictionary to `deploy` and others (no need for `get_configuration_value` in the module scripts).

dtemir commented on README.md:
```diff
@@ -8,6 +8,9 @@ Cloud Run is GCP's serverless solution for containers. With Cloud Run you can de
   Install instruction: https://cloud.google.com/sdk/docs/install and make sure all your `gcloud` components are upto data. Run `gcloud components update` to update.
 
+  * Create a project and update `cloud_run_config.json` with your `project_id`
+  * Make sure the neccassy Google APIs are enabled
```
Of course. I'll make sure to update it

jjmachan replied:
thank you!
other than that everything looks good! I've tested it locally too.

yubozhao commented:
/test-e2e-deploy sha=bdfbb1b

dtemir commented:
/test-e2e-deploy sha=61ba692

dtemir commented:
/test-e2e-deploy sha=61ba692

jjmachan commented:
/test-e2e-deploy sha=61ba692

dtemir commented:
I realized the test isn't gonna work because of the import error (will update it and try running `/test-e2e-deploy sha=61ba692` again).

I don't know if I have the permission to run `/test-e2e-deploy sha=61ba692` though?

dtemir commented:
/test-e2e-deploy sha=61ba692

jjmachan commented:
yep, it's a small import error issue but you can run the test script locally and check. If it works locally it should be good

dtemir commented:
/test-e2e-deploy sha=177afbe

dtemir commented:
was finally able to fix the `test_api.py` but can't run `/test-e2e-deploy sha=177afbe`
@bojiang could you please run the command for me? I think there's a permission issue so it doesn't run the test
```bash
$ python3 tests/test_api.py 
[2021-11-29 15:05:19,850] WARNING - Using BentoML installed in `editable` model, the local BentoML repository including all code changes will be packaged together with saved bundle created, under the './bundled_pip_dependencies' directory of the saved bundle.
temp dir /tmp/tmpvw7852d8 created!
[2021-11-29 15:05:20,482] INFO - Detected non-PyPI-released BentoML installed, copying local BentoML modulefiles to target saved bundle path..
warning: no previously-included files matching '*~' found anywhere in distribution
warning: no previously-included files matching '*.pyo' found anywhere in distribution
warning: no previously-included files matching '.git' found anywhere in distribution
warning: no previously-included files matching '.ipynb_checkpoints' found anywhere in distribution
warning: no previously-included files matching '__pycache__' found anywhere in distribution
warning: no directories found matching 'bentoml/yatai/web/dist'
no previously-included directories found matching 'e2e_tests'
no previously-included directories found matching 'tests'
no previously-included directories found matching 'benchmark'
UPDATING BentoML-0.13.1+13.gb19def03/bentoml/_version.py
set BentoML-0.13.1+13.gb19def03/bentoml/_version.py to '0.13.1+13.gb19def03'
[2021-11-29 15:05:21,261] INFO - BentoService bundle 'TestService:20211129150520_8B927D' created at: /tmp/tmpvw7852d8/saved_dir
[2021-11-29 15:05:21,272] WARNING - Saved BentoService bundle version mismatch: loading BentoService bundle create with BentoML version 0.13.1, but loading from BentoML version 0.13.1+13.gb19def03
Checking is service is up...Ok!
https://ec2-bento-deploy-test-7zfol4b6tq-uc.a.run.app/{}
Setup successful
Testing endpoint /jsonapi... passed! 
Testing endpoint /dfapi... passed! 
Testing endpoint /fileapi... passed! 
Deleted Cloud Run service [ec2-bento-deploy-test]
Deleted images
Removed /tmp/tmpvw7852d8!
```

yubozhao commented:
/test-e2e-deploy sha=177afbe

dtemir commented:
thank you! it worked
